params/params.go

```diff
@@ -129,13 +129,37 @@ type User struct {
 	LastDischarge *time.Time          `json:"last_discharge,omitempty"`
 }
 
-// SetUserRequest is request to set the details of a user.
+// SetUserRequest is a request to set the details of a user.
+// This endpoint is no longer functional.
 type SetUserRequest struct {
 	httprequest.Route `httprequest:"PUT /v1/u/:username"`
 	Username          Username `httprequest:"username,path"`
 	User              `httprequest:",body"`
 }
 
+// CreateAgentRequest is a request to add an agent.
+type CreateAgentRequest struct {
+	httprequest.Route `httprequest:"POST /v1/u"`
+	CreateAgentBody   `httprequest:",body"`
+}
+
+// CreateAgentBody holds the body of a CreateAgentRequest.
+// There must be at least one public key specified.
+type CreateAgentBody struct {
+	FullName   string              `json:"fullname"`
```

Comment:
If this is just for agent's there's no real point in being able to specify FullName and Email (GravatarID can't be trusted so should be derived from the email) SSHKeys are also probably redundant.

Reply:
Removed all but FullName, as discussed.